src/java.base/share/classes/java/util/regex/Pattern.java

```diff
@@ -820,6 +826,10 @@
      *
      * <p> Comments mode can also be enabled via the embedded flag
      * expression&nbsp;{@code (?x)}.
+     *
+     * <p> Note that comments mode ignores whitespace within a character class
+     * contained in a pattern string. Such whitespace needs to be escaped
+     * in order to be treated as if comments mode were not enabled. </p>
```

Comment:
I think this is good, but 1) it would probably be better placed in the "In this mode" paragraph above, around line 825; and 2) it's normative so it shouldn't say "Note that" (which makes it sound informative).

I'd also reword the second sentence a bit, something like

    Such whitespace needs to be escaped in order to be considered significant.